brave/browser-laptop
# Bloodhound for URL bar suggestions & move suggestions to browser process

This PR does several things:
- Makes suggestions code independent of window renderer
- Moves URL bar suggestion creating into the browser process
- Creates a wrapper lib for bloodhound autocomplete results for bookmarks and history
- Uses that wrapper lib instead of using indexOf for faster autocomplete results.
- Moves several windowActions around url bar suggestions into app actions and stops naming them set*
- Dispatcher with queryInfo of windowId specified will now always route to the browser process, before it would skip that if the window renderer was routing an app action to itself.  The browser process in that case will not re-dispatch to the window process.
- Various other perf and refactoring improvements.

Fix #7453
Fix #5878
Fix https://github.com/brave/browser-laptop/issues/8919

Submitter Checklist:

- [x] Better to not rebase in this case, changes are very cleanly defined in various small refactors leading up to using bloodhound for URL suggestions in the browser process.

Test Plan:
- Test around search options in preferences to make sure they work like they used to
- Import lots of history and bookmarks and make sure it's still snappy to get suggestions
- Make sure google.com shows up first when typing google.com
- Make sure things work at least as good as before in terms of getting results you wanted when typing.
- **QA team:** please do a full pass around URL bar suggestions, try to find things that don't work as expected but used to work before.  Please post follow ups for found things within the context of the same .300 milestone.  Please avoid posting things that have always worked that way in previous versions though.

Reviewer Checklist:

Tests


- [x] Tests should be independent and work correctly when run indivually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
